### PR TITLE
Fuzzy date picker no longer clears invalid values on blur

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Addresses #109

### CODE REVIEW

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-fuzzy-datepicker-no-clear-with-invalid-values](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-fuzzy-datepicker-no-clear-with-invalid-values)

File: src/app/public/modules/datepicker/datepicker-input-fuzzy.directive.ts [hunk 1]
```diff
@@ -340,7 +340,9 @@ export class SkyFuzzyDatepickerInputDirective
     this.onTouched();
 
     let formattedDate = this.fuzzyDateService.getStringFromFuzzyDate(this.value, this.dateFormat);
-    this.setInputElementValue(formattedDate || '');
+    if (this.control.valid) {
+      this.setInputElementValue(formattedDate || '');
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/datepicker/datepicker-input-fuzzy.directive.ts]: In the valid case - what is the point of having the `|| ''` here? If it is valid then there should always be a valid `formattedDate`, correct? Or am I missing something?

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-fuzzy-datepicker-no-clear-with-invalid-values](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-fuzzy-datepicker-no-clear-with-invalid-values)

File: src/app/public/modules/datepicker/datepicker-input-fuzzy.directive.ts (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/datepicker/datepicker-input-fuzzy.directive.ts]: The empty string allows end users to still clear the form out. If that wasn't there, we'd be sending a value of `undefined` to the `setInputElementValue()` method.

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/datepicker/datepicker-input-fuzzy.directive.ts]: Gotcha. LGTM

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-fuzzy-datepicker-no-clear-with-invalid-values](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-fuzzy-datepicker-no-clear-with-invalid-values)

**Pull request merged**